File: day04/js/part2.js
var CryptoJS = require("crypto-js");

var startdate = new Date()
secretkey="ckczppom"

cnt=0
stoploop=false
while(stoploop==false){
	cnt+=1
	mystring=secretkey+cnt.toString() //iteratively generate new string
	myhash = CryptoJS.MD5(mystring).toString()
	//console.log(myhash.substring(0,6))
	if(myhash.substring(0,6)=="000000"){
		stoploop=true
	}
}

var enddate = new Date()

console.log('Run Time:', (enddate.getTime() - startdate.getTime())/1000, 'seconds');
console.log(mystring)
console.log(myhash)
console.log("Final Answer:",cnt)

File: day04/jl/part2.jl
using MD5
using Dates

starttime=now()
secretkey="ckczppom"

cnt=0
stoploop=false
while stoploop==false
	global cnt+=1
	global mystring=secretkey*string(cnt) #iteratively generate new string
	global myhash=bytes2hex(md5(mystring)) #get the md5 hash
	if myhash[1:6]=="000000"
		global stoploop=true
    end
end

println("Run Time: $((now()-starttime).value/1000) seconds")
println(mystring)
println(myhash)
println("Final Answer:",cnt)


exit()#Stop!